### ISSUE TITLE
Player won't resize on iOS when using Metal

### ISSUE THREAD

**Issue by funniray** [cxfksword/MPVKit#17]
Hey there,

I've been playing with the demo projects provided, but I don't seem to be able to get MPV to detect the metal layer resizing... 
I've added the following to `MPVMetalViewController`
```swift
    override func viewDidLayoutSubviews() {
        let size = view.frame.size
        let scale = UIScreen.main.nativeScale
        
        metalLayer.frame = view.frame
        metalLayer.drawableSize = CGSize(width: size.width * scale, height: size.height * scale)
    }
```
I can see with the performance overlay that the metal layer has resized, however, the video content isn't modified. 

Is there any way to get resizing to work with libmpv on iOS?

**Comment by alexiscn** [comment]
try following code

```swift
    override func viewDidLayoutSubviews() {
        super.viewDidLayoutSubviews()
        
        metalLayer.frame = view.bounds
        metalLayer.contentsGravity = .resizeAspect
    }
```

**Comment by alexiscn** [comment]
You may have a look the [plex mpv fix the size of moltenvk](https://github.com/plexinc/mpv/tree/adam/play-with-sizing-on-vk) . It seems they have a solution to fix the problem. But I could not make it worked.

**Comment by funniray** [comment]
After quite a bit of trial and error -- and trying the plex fork of mpv -- I don't believe it's currently possible. It doesn't seem like anything that Plex has published so far has had fully working moltenvk resizing... I've tried multiple commits, but nothing appears to work :/

**Comment by JARMourato** [comment]
@alexiscn @funniray was any of you able to get the layer to resize correctly when device rotates / orientation changes ?

**Comment by funniray** [comment]
> was any of you able to get the layer to resize correctly when device rotates / orientation changes ?

Nope, I just ended up using openGL for iOS/macOS and then only using metal on tvOS as it's unlikely for a tv to resize.
